[BUG]: error: expression must have a constant value `constexpr int __ratio = ::cuda::ceil_div(sizeof(_Up), sizeof(uint32_t));`

### Is this a duplicate?

- [x] I confirmed there appear to be no [duplicate issues](https://github.com/NVIDIA/cccl/issues) for this bug and that I agree to the [Code of Conduct](CODE_OF_CONDUCT.md)

### Type of Bug

Compile-time Error

### Component

CUB

### Describe the bug

```
#include <algorithm>
#include <execution>
#include <vector>

using namespace std;
using namespace std::execution;

void test_case_sort_parallel_special_cases() {
    vector<int> testData;
    sort(par, testData.begin(), testData.end());
}
```
This test started failing after commit#082588c:
```
cuda/__warp/warp_shuffle.h", line 73: error: expression must have a constant value
      constexpr int __ratio = ::cuda::ceil_div(sizeof(_Up), sizeof(uint32_t));
                              ^
```

### How to Reproduce

```
nvc++ -stdpar -std=c++17  -c parallel_sort.cpp
```

### Expected behavior

Test should compile fine after the fix. 

### Reproduction link

_No response_

### Operating System

_No response_

### nvidia-smi output

_No response_

### NVCC version

_No response_

The reproducer compiles fine (but fails to link):
```
nvc++ -stdpar --c++17 -Icub -Ithrust -Ilibcudacxx/include parallel_sort.cpp
/usr/bin/x86_64-linux-gnu-ld.bfd: /usr/lib/x86_64-linux-gnu/crt1.o: in function `_start':
(.text+0x1b): undefined reference to `main'
pgacclnk: child process exit status 1: /usr/bin/ld
```
Using CCCL 830a9cbb501c0293b14470f1cee556ee75db0509 and nvc++ 26.3-0

--- a/libcudacxx/include/cuda/__cmath/ceil_div.h
+++ b/libcudacxx/include/cuda/__cmath/ceil_div.h
@@ -60,7 +60,7 @@ _CCCL_REQUIRES(::cuda::std::is_integral_v<_Tp> _CCCL_AND ::cuda::std::is_integra
   }
   else
   {
-    _CCCL_IF_CONSTEVAL
+    _CCCL_IF_CONSTEVAL_DEFAULT
     {
       const auto __res = __a1 / __b1;
       return static_cast<_Common>(__res + (__res * __b1 != __a1));

